packages/react/dialog/src/Dialog.tsx

```diff
@@ -296,28 +365,101 @@ function getState(open: boolean) {
   return open ? 'open' : 'closed';
 }
 
+const AccessibilityDevWarningsContext = React.createContext({
+  descriptionRequired: false,
+  partNames: { content: CONTENT_NAME, title: TITLE_NAME, description: DESCRIPTION_NAME },
+  docsUrl: 'https://radix-ui.com/primitives/docs/components/dialog',
+});
+
+const AccessibilityDevWarnings: React.FC<{
+  ariaLabel?: string;
+  ariaLabelledBy?: string;
+  ariaDescribedBy?: string;
+}> = (props) => {
+  const { ariaLabel, ariaLabelledBy, ariaDescribedBy } = props;
+  const context = useDialogContext('AccessibilityDevWarnings');
+  const devWarningsContext = React.useContext(AccessibilityDevWarningsContext);
+  const { partNames } = devWarningsContext;
+
+  const LABEL_WARNING = `\`${partNames.content}\` requires a label for the component to be accessible for screen reader users.
```

Comment:
Mostly copied component from what was in `AlertDialog` but brought the warnings inline here so I could add the localised names from context.